llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/base.py

```diff
@@ -141,6 +147,7 @@ def __init__(
             model=model,
             examples=examples,
             iscode=iscode,
+            safety_settings=safety_settings,
```

Comment:
I might be blind, but I think this is not used anywhere?

Reply:
Hey @logan-markewich  - So I was trying to test the changes with my code in the PR description but I was not getting the `safety_settings`  initialized and it was being returned as `None`. After adding the same to `super()` I was able to get it initialized and test cases were passing. I could have added the test module but it requires credntials, so decided not to add it.

Reply:
Assert fails I remove this:

```
FAILED just_testing.py::test_vertex_initialization - AssertionError: assert None == {'HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT': 'HarmBlockThreshold.BLOCK_LOW_AND_ABOVE', 'HarmCategory.HARM_CATEGORY_HATE_SPEECH': 'HarmBlockThreshold.BLOCK_LOW_AND_ABOVE'}
```

Reply:
Right we need it. But the actually safety settings aren't being passed to the LLM anywhere?

Reply:
Yes, It needs to be passed as below:

```
  llm = Vertex(
        model="gemini-pro",
        project='GCP_PROJECT_ID',
        safety_settings=safety_settings,
        location="GCP_PROJECT_LOCATION",
    )
   ```

Reply:
I think, The safety_settings value is  None because it's not being passed to the parent class LLM during initialization. When I call super().__init__() in the Vertex class's __init__ method, I am able to initializing the parent class LLM. If safety_settings is not included in this call, it won't be set in the parent class.

Reply:
@logan-markewich If I am not wrong this is supposed to be merged automatically? IF not could you please help with the same, I am waiting on this change and one more to be used from the latest version. Thanks!

Reply:
@akumar-b6i sorry, I don't think my point is getting across.

This PR doesn't seem to actually be setting the safety settings to the vertex client. It's not used in either the client constructor (i.e. `.from_pretrianed(..)`), nor is it passed to the model through kwargs when getting a response. Its just attached to the LLM class object and never used

Reply:
Thanks @logan-markewich, and sorry, It took me some time. I made some further changes and tested the same by passing some default safety_setting and a question, which was blocked initially but later changed the safety settings to BLOCK_NONE and now I can retrieve a response using Gemini-pro. Hope this helps!

Reply:
Awesome, thanks